From 660330e611e94235f0be55d5dde60f3d2753fe4c Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:40:13 +0800
Subject: [PATCH 20/25] GSI Android 17 compatibility:
 0020-Add-a-prop-to-disable-AUDIO_OUTPUT_FLAG_FAST-from-au.patch

Change-Id: I16f641ee6961b62451af876d85b2e0b7865460e9
---
 .../audiopolicy/common/managerdefinitions/src/Serializer.cpp  | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/services/audiopolicy/common/managerdefinitions/src/Serializer.cpp b/services/audiopolicy/common/managerdefinitions/src/Serializer.cpp
index 77a96b4..26edcaa 100644
--- a/services/audiopolicy/common/managerdefinitions/src/Serializer.cpp
+++ b/services/audiopolicy/common/managerdefinitions/src/Serializer.cpp
@@ -504,6 +504,10 @@ std::variant<status_t, MixPortTraits::Element> PolicySerializer::deserialize<Mix
             // use DEEP_BUFFER+FAST flag combo to indicate the spatializer output profile
             uint32_t intFlags =
                     OutputFlagConverter::maskFromString(flags, mFlagsSeparator.c_str());
+            bool ignore_fast = property_get_bool("persist.sys.phh.disable_fast_audio", false);
+            if (ignore_fast)
+                intFlags &= ~AUDIO_OUTPUT_FLAG_FAST;
+
             if (intFlags == (AUDIO_OUTPUT_FLAG_FAST | AUDIO_OUTPUT_FLAG_DEEP_BUFFER)) {
                 intFlags = AUDIO_OUTPUT_FLAG_SPATIALIZER;
             }
